File: src/MixNetFx/FacFunctions/Dialogs/IncentiveWageRelated/GetFrmIncentiveWageCalc.vb
Imports Facesso.Data
Imports System.Data.SqlClient
Imports System.Windows.Forms

Public Class GetFrmIncentiveWageCalc
    Inherits FacessoFunctionBase

    Public Overrides ReadOnly Property RolePermission() As IRolePermissionInfo
        Get
            Return New FacessoRolePermissionInfo(ClearanceLevel.PerformAccounting)
        End Get
    End Property

    Public Overrides ReadOnly Property RolePermissionViolationMessage() As String
        Get
            Return My.Resources.IncentiveWageCalculation_DeniedDueToRole
        End Get
    End Property

    Public Function ShowDialog() As DialogResult

        Dim locFH As GetFrmIncentiveWageCalc = FunctionHandler(Of GetFrmIncentiveWageCalc).GetFunctionInstance
        If locFH Is Nothing Then Return Nothing

        Dim locFrmIncentiveWageCalc As New frmIncentiveWageCalc
        Return locFrmIncentiveWageCalc.ShowDialog
    End Function
End Class

File: src/NetFxCSharp/Facesso.Functions/Dialogs/IncentiveWageRelated/GetFrmIncentiveWageCalc.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public class GetFrmIncentiveWageCalc : FacessoFunctionBase
    {
        public override IRolePermissionInfo RolePermission
        {
            get
            {
                return new FacessoRolePermissionInfo(ClearanceLevel.PerformAccounting);
            }
        }

        public override string RolePermissionViolationMessage
        {
            get
            {
                return Facesso.Functions.My.Resources.IncentiveWageCalculation_DeniedDueToRole;
            }
        }

        public DialogResult ShowDialog()
        {
            GetFrmIncentiveWageCalc locFH = FunctionHandler<GetFrmIncentiveWageCalc>.GetFunctionInstance();
            if (locFH == null)
            {
                return default(System.Windows.Forms.DialogResult);
            }

            frmIncentiveWageCalc locFrmIncentiveWageCalc = new frmIncentiveWageCalc();
            return locFrmIncentiveWageCalc.ShowDialog();
        }
    }
}